src/Swarm/TUI/Controller.hs

```diff
@@ -189,7 +189,7 @@ handleMainEvent s = \case
     continue s
   Key V.KEsc
     | isJust (s ^. uiState . uiError) -> continue $ s & uiState . uiError .~ Nothing
-    | isJust (s ^. uiState . uiModal) -> continue $ s & uiState . uiModal .~ Nothing
+    | isJust (s ^. uiState . uiModal) -> continue $ s & maybeUnpause s . (uiState . uiModal .~ Nothing)
```

Comment:
We need to also call `resetLastFrameTime` here; see the comment below next to the other use of `maybeUnpause`.  Probably it would be better to refactor them both into the same function, so we can't call only one without the other.

Reply:
Good catch. I've proposed a fix separately in #452